File: flutter_zoom_videosdk-2.4.0/android/src/main/java/com/flutterzoom/videosdk/FlutterZoomVideoSdkChatHelper.java
package com.flutterzoom.videosdk;

import android.app.Activity;

import androidx.annotation.NonNull;

import com.flutterzoom.videosdk.convert.FlutterZoomVideoSdkChatPrivilegeType;
import com.flutterzoom.videosdk.convert.FlutterZoomVideoSdkErrors;

import java.util.Map;

import io.flutter.plugin.common.MethodCall;
import io.flutter.plugin.common.MethodChannel;
import us.zoom.sdk.ZoomVideoSDK;
import us.zoom.sdk.ZoomVideoSDKChatHelper;
import us.zoom.sdk.ZoomVideoSDKChatPrivilegeType;
import us.zoom.sdk.ZoomVideoSDKErrors;
import us.zoom.sdk.ZoomVideoSDKUser;

public class FlutterZoomVideoSdkChatHelper {

    private Activity activity;

    FlutterZoomVideoSdkChatHelper(Activity activity) {
        this.activity = activity;
    }

    private ZoomVideoSDKChatHelper getChatHelper() {
        ZoomVideoSDKChatHelper chatHelper = null;
        try {
            chatHelper = ZoomVideoSDK.getInstance().getChatHelper();
            if (chatHelper == null) {
                throw new Exception("No Chat Helper Available");
            }
        } catch (Exception e) {
            e.printStackTrace();
        }
        return chatHelper;
    }

    public void isChatDisabled(@NonNull MethodChannel.Result result) {
        ZoomVideoSDKChatHelper chatHelper = getChatHelper();
        result.success(chatHelper.isChatDisabled());
    }

    public void isPrivateChatDisabled(@NonNull MethodChannel.Result result) {
        ZoomVideoSDKChatHelper chatHelper = getChatHelper();
        result.success(chatHelper.isPrivateChatDisabled());
    }

    public void sendChatToUser(@NonNull MethodCall call, @NonNull MethodChannel.Result result) {
        Map<String, Object> param = call.arguments();
        String userId = (String) param.get("userId");
        String message = (String) param.get("message");

        ZoomVideoSDKChatHelper chatHelper = getChatHelper();
        if (chatHelper.isChatDisabled() || chatHelper.isPrivateChatDisabled()) {
            return;
        }
        ZoomVideoSDKUser user = FlutterZoomVideoSdkUser.getUser(userId);

        activity.runOnUiThread(new Runnable() {
            @Override
            public void run() {
                result.success(FlutterZoomVideoSdkErrors.valueOf(chatHelper.sendChatToUser(user, message)));
            }
        });
    }

    public void sendChatToAll(@NonNull MethodCall call, @NonNull MethodChannel.Result result) {
        Map<String, Object> args = call.arguments();
        String message = (String) args.get("message");

        ZoomVideoSDKChatHelper chatHelper = getChatHelper();
        if (chatHelper.isChatDisabled()) {
            return;
        }
        activity.runOnUiThread(new Runnable() {
            @Override
            public void run() {
                result.success(FlutterZoomVideoSdkErrors.valueOf(chatHelper.sendChatToAll(message)));
            }
        });
    }

    public void deleteChatMessage(@NonNull MethodCall call, @NonNull MethodChannel.Result result) {
        Map<String, Object> args = call.arguments();
        String msgId = (String) args.get("msgId");
        if (msgId == null || msgId.isEmpty()) {
            result.error(FlutterZoomVideoSdkErrors.valueOf(ZoomVideoSDKErrors.Errors_Invalid_Parameter), "msgId is null or empty", null);
        }

        ZoomVideoSDKChatHelper chatHelper = getChatHelper();
        if (chatHelper.isChatDisabled()) {
            return;
        }
        activity.runOnUiThread(new Runnable() {
            @Override
            public void run() {
                result.success(FlutterZoomVideoSdkErrors.valueOf(chatHelper.deleteChatMessage(msgId)));
            }
        });
    }

    public void canChatMessageBeDeleted(@NonNull MethodCall call, @NonNull MethodChannel.Result result) {
        Map<String, Object> args = call.arguments();
        String msgId = (String) args.get("msgId");
        if (msgId == null || msgId.isEmpty()) {
            result.error(FlutterZoomVideoSdkErrors.valueOf(ZoomVideoSDKErrors.Errors_Invalid_Parameter), "msgId is null or empty", null);
        }

        ZoomVideoSDKChatHelper chatHelper = getChatHelper();
        if (chatHelper.isChatDisabled()) {
            return;
        }
        result.success(chatHelper.canChatMessageBeDeleted(msgId));
    }

    public void changeChatPrivilege(@NonNull MethodCall call, @NonNull MethodChannel.Result result) {
        Map<String, Object> args = call.arguments();
        String privilege = (String) args.get("privilege");
        if (privilege == null || privilege.isEmpty()) {
            result.error(FlutterZoomVideoSdkErrors.valueOf(ZoomVideoSDKErrors.Errors_Invalid_Parameter), "privilege is null or empty", null);
        }

        ZoomVideoSDKChatHelper chatHelper = getChatHelper();
        if (chatHelper.isChatDisabled()) {
            return;
        }
        ZoomVideoSDKChatPrivilegeType type = FlutterZoomVideoSdkChatPrivilegeType.typeOf(privilege);
        result.success(FlutterZoomVideoSdkErrors.valueOf(chatHelper.changeChatPrivilege(type)));
    }

    public void getChatPrivilege(@NonNull MethodChannel.Result result) {

        ZoomVideoSDKChatHelper chatHelper = getChatHelper();
        if (chatHelper.isChatDisabled()) {
            return;
        }
        result.success(FlutterZoomVideoSdkChatPrivilegeType.valueOf(chatHelper.getChatPrivilege()));
    }

}

File: flutter_zoom_videosdk-2.4.0/ios/Classes/FlutterZoomVideoSdkChatHelper.m
#import "FlutterZoomVideoSdkChatHelper.h"
#import "FlutterZoomVideoSdkUser.h"
#import "JSONConvert.h"

@implementation FlutterZoomVideoSdkChatHelper

-(void) isChatDisabled: (FlutterResult) result {
    ZoomVideoSDKChatHelper* chatHelper = [[ZoomVideoSDK shareInstance] getChatHelper];
    if ([chatHelper IsChatDisabled]) {
        result(@YES);
    } else {
        result(@NO);
    }
}

-(void) isPrivateChatDisabled: (FlutterResult) result {
    ZoomVideoSDKChatHelper* chatHelper = [[ZoomVideoSDK shareInstance] getChatHelper];
    if ([chatHelper IsPrivateChatDisabled]) {
        result(@YES);
    } else {
        result(@NO);
    }
}

-(void) sendChatToUser: (FlutterMethodCall *)call withResult:(FlutterResult) result {
    ZoomVideoSDKChatHelper* chatHelper = [[ZoomVideoSDK shareInstance] getChatHelper];

    if (chatHelper.IsChatDisabled == NO && chatHelper.IsPrivateChatDisabled == NO) {
        NSString* userId = call.arguments[@"userId"];
        if ([userId length] == 0) {
            result([FlutterError errorWithCode:[[JSONConvert ZoomVideoSDKErrorValuesReversed] objectForKey: @(Errors_Invalid_Parameter)]
                                       message:@"UserId is empty"
                                       details:nil]);
        }
        ZoomVideoSDKUser *user = [FlutterZoomVideoSdkUser getUser:userId];
        if (user != nil) {
            dispatch_async(dispatch_get_main_queue(), ^{
                result([[JSONConvert ZoomVideoSDKErrorValuesReversed] objectForKey: @([chatHelper SendChatToUser:user Content: call.arguments[@"message"]])]);
            });
        } else {
            result([FlutterError errorWithCode:[[JSONConvert ZoomVideoSDKErrorValuesReversed] objectForKey: @(Errors_Invalid_Parameter)]
                                       message:@"User not found"
                                       details:nil]);
        }
    }
}

-(void) sendChatToAll: (FlutterMethodCall *)call withResult:(FlutterResult) result {
    ZoomVideoSDKChatHelper* chatHelper = [[ZoomVideoSDK shareInstance] getChatHelper];

    if (chatHelper.IsChatDisabled == NO) {
        dispatch_async(dispatch_get_main_queue(), ^{
            result([[JSONConvert ZoomVideoSDKErrorValuesReversed] objectForKey: @([chatHelper SendChatToAll: call.arguments[@"message"]])]);
        });
    }
}

-(void) deleteChatMessage: (FlutterMethodCall *)call withResult:(FlutterResult) result {
    ZoomVideoSDKChatHelper* chatHelper = [[ZoomVideoSDK shareInstance] getChatHelper];

    if (chatHelper.IsChatDisabled == NO) {
        dispatch_async(dispatch_get_main_queue(), ^{
            result([[JSONConvert ZoomVideoSDKErrorValuesReversed] objectForKey: @([chatHelper deleteChatMessage: call.arguments[@"msgId"]])]);
        });
    }
}

-(void) canChatMessageBeDeleted: (FlutterMethodCall *)call withResult:(FlutterResult) result {
    ZoomVideoSDKChatHelper* chatHelper = [[ZoomVideoSDK shareInstance] getChatHelper];

    if (chatHelper.IsChatDisabled == NO) {
        dispatch_async(dispatch_get_main_queue(), ^{
            if ([chatHelper canChatMessageBeDeleted: call.arguments[@"msgId"]]) {
                result(@YES);
            } else {
                result(@NO);
            }
        });
    }
}

-(void) changeChatPrivilege: (FlutterMethodCall *)call withResult:(FlutterResult) result {
    ZoomVideoSDKChatHelper* chatHelper = [[ZoomVideoSDK shareInstance] getChatHelper];
    NSString* privilege = call.arguments[@"privilege"];
    if ([privilege length] == 0) {
        result([FlutterError errorWithCode:[[JSONConvert ZoomVideoSDKErrorValuesReversed] objectForKey: @(Errors_Invalid_Parameter)]
                                   message:@"privilege is empty"
                                   details:nil]);
    }
    ZoomVideoSDKChatPrivilegeType type = [JSONConvert ZoomVideoSDKChatPrivilegeType: privilege];
    
    if (chatHelper.IsChatDisabled == NO) {
        dispatch_async(dispatch_get_main_queue(), ^{
            result([[JSONConvert ZoomVideoSDKErrorValuesReversed] objectForKey: @([chatHelper changeChatPrivilege: type])]);
        });
    }
}

-(void) getChatPrivilege: (FlutterResult) result {
    ZoomVideoSDKChatHelper* chatHelper = [[ZoomVideoSDK shareInstance] getChatHelper];

    if (chatHelper.IsChatDisabled == NO) {
        dispatch_async(dispatch_get_main_queue(), ^{
            result([[JSONConvert ZoomVideoSDKChatPrivilegeTypeValuesReversed] objectForKey: @([chatHelper getChatPrivilege])]);
        });
    }
}

@end
